### PR TITLE
[MM-223] Fixing Post Permissions Access

### PR DESCRIPTION
**Pull request by vicky-demansol**
#### Summary
Prevent user post zoom call if channel permissions make member with read-only permissions

#### Ticket Link
https://github.com/mattermost/mattermost-plugin-zoom/issues/223

### CODE REVIEW

**Comment by mickmister** [comment]
Please note that there are CI failures occurring in this PR, including tests failing:

image

File: server/http.go [hunk 1]
```diff
@@ -303,6 +303,10 @@ func (p *Plugin) postMeeting(creator *model.User, meetingID int, channelID strin
 		topic = defaultMeetingTopic
 	}
 
+	if !p.API.HasPermissionToChannel(creator.Id, channelID, model.PERMISSION_CREATE_POST) {
+		return errors.New("this channel is not accessible, you might not have permissions to write in this channel. Contact the administrator of this channel to find out if you have access permissions")
```
Comment (mickmister) [inline server/http.go:307]: @vicky-demansol We still need a unit test that covers the inside of this if statement

Comment (vicky-demansol) [reply server/http.go:307]: I was add the unit test but the server response will return http code = 500, this response could be it returned as "forbidden" or "unauthorized"

**Comment by emilyacook** [comment]
Thanks for participating in Hacktoberfest! You can claim your sticker set here: https://get.printfection.com/hacktober21/4144583267 @vicky-demansol

File: server/http.go (hunk 1 above)
Comment (mickmister) [reply server/http.go:307]: Hi @vicky-demansol, please make sure you mention the person you're replying to on GitHub issues/PRs, so we can stay up-to-date on discussions like this.

> I was add the unit test but the server response will return http code = 500, this response could be it returned as "forbidden" or "unauthorized"

Can we create a unit test just to cover that this line will run? If it take a lot of restructure to accomplish changing the response code, let's just make sure this error occurs in general.

Comment (vicky-demansol) [reply server/http.go:307]: @micmister i was trying, if we want to get http code error need to use appError but i don't think this function can use in general, but this code on code coverage was covered

**Comment by dipak-demansol** [comment]
@hanzei i tested this and its working but i have some question so after a call with @DHaussermann i will add the final review at the end of the day.

**Review by dipak-demansol — APPROVED** [review]
LGTM

**Pull request merged**